src/Query/TemplateQuery.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+/*
+ * This file is part of the ONGR package.
+ *
+ * (c) NFQ Technologies UAB <info@nfq.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace ONGR\ElasticsearchDSL\Query;
+
+use ONGR\ElasticsearchDSL\BuilderInterface;
+
+/**
+ * Represents Elasticsearch "template" query.
+ *
+ * @link https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-template-query.html
+ */
+class TemplateQuery implements BuilderInterface
+{
+    /**
+     * @var string
+     */
+    private $inline;
+
+    /**
+     * @var array
+     */
+    private $params;
+
+    /**
+     * @param string $inline A template of the query
+     * @param array  $params Parameters to insert into template
+     */
+    public function __construct($inline, array $params)
```

Comment:
Missing `file`. This is kind of difficult situation, which one we should leave. IMO there should be both. Like this:

```php
public function __construct($file = null, $inline = null, array $params = [])
```